### ISSUE TITLE
Using Package version with a semver pre-release label like 1.0.0-alpha1 fails reconciling

### ISSUE THREAD

**Issue by trisberg** [carvel-dev/kapp-controller#116]
**What steps did you take:**

Used a version for the Package resource that included an `alpha1` pre-release label

```
apiVersion: package.carvel.dev/v1alpha1
kind: Package
metadata:
  name: tss-app.tanzu.vmware.com.1.0.0-alpha1
spec:
  publicName: tss-app.tanzu.vmware.com
  version: 1.0.0-alpha1
  template:
    spec:
...
```

**What happened:**

The InstalledImage referencing this package fails reconciliation:

```
status:
  conditions:
  - message: Expected to find at least one version, but did not
    status: "True"
    type: ReconcileFailed
  friendlyDescription: 'Reconcile failed: Expected to find at least one version, but
    did not'
```

**What did you expect:**

That it would work same way as if I removed the pre-release label and used version of 1.0.0

**Anything else you would like to add:**

**Environment:**

- kapp Controller version (execute `kubectl get deployment -n kapp-controller kapp-controller -o yaml` and the annotation is `kbld.k14s.io/images`):
    ```
    kbld.k14s.io/images: |
      - Metas:
        - Path: /Users/embp/go/src/github.com/vmware-tanzu/carvel-kapp-controller
          Type: local
        - Dirty: true
          RemoteURL: git@github.com:vmware-tanzu/carvel-kapp-controller.git
          SHA: e6755bba67e2d176698f198eb22efd8430dfc4b3
          Type: git
        URL: index.docker.io/k8slt/kapp-controller@sha256:4c1433f8b0fbc0f49819d4703edd196f2f0a99907254b2291a8776b1c3665c98
    ```

- Kubernetes version (use `kubectl version`):
    ```
   Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
    Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.15-gke.800", GitCommit:"2e625a55a420ded12d04226c63c4cf415efc9cdb", GitTreeState:"clean", BuildDate:"2021-01-25T09:17:48Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}
    ```

**Comment by danielhelfand** [comment]
The error seems to be coming from vendir: 
https://github.com/vmware-tanzu/carvel-vendir/blob/bf0685a51eae291131fe5d1475ac0a6eceeb3c1e/pkg/vendir/versions/v1alpha1/selector.go#L23

It is called from installedpkg.go: https://github.com/vmware-tanzu/carvel-kapp-controller/blob/4234a4b21577a792548c6a6405a4e84054e7f466/pkg/installedpkg/installedpkg.go#L154.

**Comment by danielhelfand** [comment]
In digging a bit further into this, it looks like kapp-controller is not providing the proper config by default for an exact match on a PackageRef version:

https://github.com/vmware-tanzu/carvel-kapp-controller/blob/4234a4b21577a792548c6a6405a4e84054e7f466/pkg/installedpkg/installedpkg.go#L134

In this case, we need the Prereleases to be non nil since `HighestConstrainedVersion` will [not include prerelease versions otherwise](https://github.com/vmware-tanzu/carvel-vendir/blob/bf0685a51eae291131fe5d1475ac0a6eceeb3c1e/pkg/vendir/versions/v1alpha1/semvers.go#L75): 

```go
semverConfig = &versions.VersionSelectionSemver{Constraints: ip.model.Spec.PkgRef.Version, 
Prereleases: &versions.VersionSelectionSemverPrereleases{}}
```

**Comment by danielhelfand** [comment]
Added the fix here: https://github.com/vmware-tanzu/carvel-kapp-controller/blob/5839f6167ae220289bbe89a533b141e6cdd1a1df/pkg/installedpkg/installedpkg.go#L137

This should be available in next alpha release candidate.

**Comment by ewrenn8** [comment]
Fixed in `v0.17.0-alpha.2` via this commit a561615ead184e5937d4e857e87f1bd338b84b74
